helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -133,12 +134,11 @@ public void process(ClusterEvent event) throws Exception {
 
   private void addSwapInInstancesToBestPossibleState(Map<String, Resource> resourceMap,
       BestPossibleStateOutput bestPossibleStateOutput, ResourceControllerDataProvider cache) {
```

Comment:
Should copy the exact state of the old node except for top state if only one is allowed.

Reply:
We will not copy ERROR replicas and will copy OFFLINE replicas as OFFLINE.